apps/backend/src/app/descart/descart.service.ts

```diff
@@ -248,10 +248,12 @@ export class DescartService {
   async createPurchase(userId: number, body: CreatePurchaseDto): Promise<Purchase> {
     const today = new Date();
     let numItems = 0;
+    const itemIds: Set<number> = new Set();
     body.products.map((product: ProductDto) => {
       numItems += product.quantity;
+      itemIds.add(product.id)
```

Comment:
change this to `if (product.id) itemIds.add(product.id)`, then go ahead and merge.

(custom products don't have an id so this crashes when a purchase has one)